test/unit/simplifyAST.test.js

```diff
@@ -73,17 +73,45 @@ describe('simplifyAST', function () {
   it('should simplify a basic structure with array args', function () {
     expect(simplifyAST(parse(`
       {
-        luke: human(id: ["1000", "1003"]) {
+        user(units: ["1", "2", ["3", ["4"], [["5"], "6"], "7"]]) {
```

Comment:
almost yes, i still prefer you add a test if it does something slightly different, unit tests are cheap :)